service/test/ServiceIOGroupTest.cpp

```diff
@@ -192,6 +223,39 @@ TEST_F(ServiceIOGroupTest, write_control)
     }
 }
 
+TEST_F(ServiceIOGroupTest, write_control_exception)
+{
+    // !is_valid_control(control_name)
+    GEOPM_EXPECT_THROW_MESSAGE(
+        m_serviceio_group->write_control("NUM_VACCUM_TUBES", 4, 0, 7.0),
```

Comment:
```suggestion
        m_serviceio_group->write_control("NUM_VACUUM_TUBES", 4, 0, 7.0),
```

I know it's a nit, but it's spelled "vacuum".  This does not affect the functionality.